### PR TITLE
feat(events): store short timezone label on Buzz Event

### PR DESCRIPTION
**Pull request by harshtandiya**
## Why

`time_zone` stores IANA names ("Asia/Dubai") which read poorly next to event times, so the dashboard hid the timezone entirely. Public pages need a short label like "GST", "IST", or "GMT+4".

## What

- **`buzz/utils.py`** — `get_time_zone_label(time_zone, reference_datetime)`:
  1. tzdata abbreviation when it ships a real one (IST, EST, CET — DST-aware: NY July event → EDT)
  2. curated `TIMEZONE_ABBREVIATIONS` map (34 entries) for zones tzdata reduced to numeric offsets in 2017 (GST, AST, ICT, NPT…)
  3. formatted GMT offset fallback ("GMT+5:45")
- **Buzz Event** — new read-only `time_zone_label` Data field, set in `validate` at the event start datetime; guarded against missing dates (validate runs before the mandatory check)
- **APIs** — guest booking data + custom form data expose the label
- **Dashboard** — event header shows `(GST)` style label again
- **Patch** — backfills existing events (`db_set`, no hooks)

## Tests

9 new tests (TDD): tzdata direct, DST variant by reference date, curated map, GMT fallback incl. half-hour and negative offsets, empty/invalid zones, and controller persistence on insert/update/clear/DST. Full app suite green (220 tests).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: buzz/patches/set_time_zone_label_for_existing_events.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+def execute():
+	events = frappe.get_all(
+		"Buzz Event",
+		filters={"time_zone": ("is", "set")},
+		fields=["name", "time_zone", "start_date", "start_time"],
+	)
+	for event in events:
+		event_start = get_datetime(f"{event.start_date} {event.start_time}")
+		label = get_time_zone_label(event.time_zone, event_start)
+		frappe.db.set_value("Buzz Event", event.name, "time_zone_label", label, update_modified=False)
```
[the remark this answers is not in the document]
Comment (harshtandiya) [reply buzz/patches/set_time_zone_label_for_existing_events.py:20]: Fixed in 1318210 — patch filters now require start_date and start_time set, matching the controller guard. Regression test added (legacy row with null start_time no longer aborts the run).

File: buzz/utils.py [hunk 2]
```diff
@@ -188,3 +192,78 @@ def generate_ics_file(event_doc, attendee_email: str):
...
+	"Asia/Novosibirsk": "NOVT",
+	"Asia/Qatar": "AST",
+	"Asia/Riyadh": "AST",
+	"Asia/Saigon": "ICT",
+	"Asia/Tashkent": "UZT",
+	"Asia/Tehran": "IRST",
+	"Asia/Yekaterinburg": "YEKT",
+	"Atlantic/Azores": "AZOT",
+	"Atlantic/Cape_Verde": "CVT",
+	"Europe/Istanbul": "TRT",
```
[the remark this answers is not in the document]
Comment (harshtandiya) [reply buzz/utils.py:239]: Added America/Nuuk and Asia/Ho_Chi_Minh entries in 1318210. Note the field's options come from the Zoom-supported list in buzz_event.js (legacy names only), so the new spellings aren't reachable from the picker today — entries added for robustness, tested.

File: buzz/utils.py [hunk 3]
```diff
@@ -188,3 +192,78 @@ def generate_ics_file(event_doc, attendee_email: str):
...
+	"""
+	if not time_zone:
+		return ""
+
+	try:
+		zone = ZoneInfo(time_zone)
+	except (ZoneInfoNotFoundError, ValueError):
+		return ""
+
+	moment = (reference_datetime or now_datetime()).replace(tzinfo=zone)
```
[the remark this answers is not in the document]
Comment (harshtandiya) [reply buzz/utils.py]: Fixed in 1318210 — aware inputs now go through astimezone() (correct conversion) instead of replace(); naive inputs unchanged. Test covers the DST-boundary case (05:30 UTC on 2026-11-01 → EDT).
